src/devices/MettlerToledo/MettlerToledoDevice.cs

```diff
@@ -0,0 +1,362 @@
+
+    /// <summary>
+    /// Provides an interface for interacting with MT-SICS devices, like scales.
+    /// </summary>
+    public class MettlerToledoDevice : IDisposable
+    {
+        // TODO: we need to test on a real scale to determine what the response for other weights are when they are set to Unit 1
+        private static readonly (string mtAbbreviation, MettlerToledoWeightUnit mtUnit, UnitsNet.Units.MassUnit unitsNetUnit)[] _unitMapping =
+        {
+            ("g", MettlerToledoWeightUnit.Grams, UnitsNet.Units.MassUnit.Gram),
+            ("kg", MettlerToledoWeightUnit.Kilograms, UnitsNet.Units.MassUnit.Kilogram),
+            ("mg", MettlerToledoWeightUnit.Milligrams, UnitsNet.Units.MassUnit.Milligram)
+        };
+
+        private SerialPort _serialPort;
+        private bool _shouldDisposeSerialPort = true;
+
+        /// <summary>
+        /// Called when the scale reports a new weight, outside of the response to a command. Typically this is due to a call to <see cref="SubscribeToWeightAtIntervals()"/> or <see cref="SubscribeToWeightChangeEvents()"/>
+        /// </summary>
+        public event WeightUpdatedHandler WeightUpdated;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MettlerToledoDevice"/> class.
+        /// </summary>
+        public MettlerToledoDevice(string port, int baudrate = 9600)
+        {
+            _serialPort = new SerialPort(port, baudrate, Parity.None);
+            _serialPort.Encoding = Encoding.ASCII;
+            _serialPort.NewLine = "\r\n";
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MettlerToledoDevice"/> class.
+        /// </summary>
+        /// <param name="serialPort">Serial port to use</param>
+        /// <param name="shouldDispose">Should the serial port be disposed by this class?</param>
+        public MettlerToledoDevice(SerialPort serialPort, bool shouldDispose = true)
+        {
+            _serialPort = serialPort;
+            _serialPort.Encoding = Encoding.ASCII;
+            _serialPort.NewLine = "\r\n";
+            _shouldDisposeSerialPort = shouldDispose;
+        }
+
+        /// <summary>
+        /// Retrieves the current stable weight from the scale.
+        /// </summary>
+        /// <returns>Current weight</returns>
+        public MettlerToledoWeightReading GetStableWeight() => GetWeightReading(false);
+
+        /// <summary>
+        /// Retrieves the last internal weight value (stable or dynamic) before receipt of this command.
+        /// </summary>
+        /// <returns>Current weight reading</returns>
+        public MettlerToledoWeightReading GetImmediateWeight() => GetWeightReading(true);
+
+        /// <summary>
+        /// Opens a connection to the MT-SICS scale.
+        /// </summary>
+        /// <exception cref="UnauthorizedAccessException"></exception>
+        public void Open()
```

Comment:
Should that be hidden? You may open when you create and Close when you dispose. Any reason to leave those 2 public?